integration-tests/types/action.ts

```diff
@@ -0,0 +1,66 @@
+// import App, { ActionConstraints } from '../../src/App';
+// import { MessageAction } from '../../src/types';
+import { App } from '@slack/bolt';
+
+const app = new App({ token: 'TOKEN', signingSecret: 'Signing Secret' });
+
+// calling action method with incorrect an type constraint value should not work
+// $ExpectError
+app.action({ type: 'Something wrong' }, ({ action }) => {
+  return action;
+});
+
+/*
+// Should error because message_action doesn't have type action_id
+// $ Expect Error
+app.action({ type: 'message_action', action_id: 'dafasf' }, ({ action }) => {
+    // NOT WORKING
+    return action;
+});
+*/
+
+// Action in listner should be - MessageAction
+// $ExpectType void
+app.action({ type: 'message_action' }, ({
+  action, // $ExpectType MessageAction
+ }) => {
+  return action;
+});
+
+// $ExpectType void
+app.action({ type: 'block_actions' }, ({
+  action, // $ExpectType BlockElementAction
```

Comment:
I think this actually resolves to `BlockAction<BlockElementAction>`. have we run this to make sure its right?

Reply:
The current one is passing. 

When I enter your suggestion, I get the following error:

```
ERROR: 30:3  expect  Expected type to be:
  BlockAction<BlockElementAction>
got:
  BlockElementAction
```

Reply:
oh, i think i misinterpreted the wrappers. 🤦‍♂